File: class10/python/GQS1001-MatrizRisco.py
#!/usr/bin/env python3
"""
GQS1001 - Calculadora de Exposição a Risco (Probabilidade × Impacto).
Uso: python3 GQS1001-MatrizRisco.py
"""


def calcular_risco(probabilidade: int, impacto: int) -> int:
    """Risco = Probabilidade × Impacto."""
    return probabilidade * impacto


def classificar_risco(risco: int) -> str:
    if risco >= 15:
        return "🔴 Crítico"
    elif risco >= 10:
        return "🟠 Alto"
    elif risco >= 5:
        return "🟡 Médio"
    else:
        return "🟢 Baixo"


def main():
    print("=" * 60)
    print("  GQS1001 - MATRIZ DE RISCO (Probabilidade × Impacto)")
    print("=" * 60)

    print("\nEscala de Probabilidade e Impacto (1 a 5):")
    print("  1 = Muito Baixo | 2 = Baixo | 3 = Médio | 4 = Alto | 5 = Muito Alto")

    funcionalidades = [
        ("Checkout/Pagamento", 3, 5),
        ("Carrinho de Compras", 4, 4),
        ("Catálogo de Produtos", 3, 2),
        ("Cadastro de Usuário", 2, 3),
        ("Busca/Filtros", 4, 2),
    ]

    print(f"\n{'Funcionalidade':<25} {'Prob.':<8} {'Imp.':<8} {'Risco':<8} {'Classificação':<15}")
    print("-" * 65)
    for nome, prob, imp in funcionalidades:
        risco = calcular_risco(prob, imp)
        classificacao = classificar_risco(risco)
        print(f"{nome:<25} {prob:<8} {imp:<8} {risco:<8} {classificacao:<15}")

    print("\n--- Priorização de Testes ---")
    print("  Crítico: Teste exaustivo, automação prioritária")
    print("  Alto: Teste extensivo")
    print("  Médio: Teste aprofundado")
    print("  Baixo: Teste padrão")


if __name__ == "__main__":
    main()

File: class10/java/GQS1001-MatrizRisco.java
/**
 * GQS1001 - Calculadora de Exposição a Risco (Probabilidade x Impacto).
 * Reimplementação em Java da matriz de risco usada para priorização de testes.
 *
 * Compilação e execução:
 *   javac GQS1001-MatrizRisco.java
 *   java -cp . GQS1001_MatrizRisco
 */
import java.util.*;

/** Representa uma funcionalidade avaliada na matriz de risco. */
class Funcionalidade {
    String nome;
    int probabilidade;
    int impacto;

    Funcionalidade(String nome, int probabilidade, int impacto) {
        this.nome = nome;
        this.probabilidade = probabilidade;
        this.impacto = impacto;
    }
}

class GQS1001_MatrizRisco {

    /** Risco = Probabilidade x Impacto. */
    static int calcularRisco(int probabilidade, int impacto) {
        return probabilidade * impacto;
    }

    static String classificarRisco(int risco) {
        if (risco >= 15) {
            return "🔴 Crítico";
        } else if (risco >= 10) {
            return "🟠 Alto";
        } else if (risco >= 5) {
            return "🟡 Médio";
        } else {
            return "🟢 Baixo";
        }
    }

    public static void main(String[] args) {
        System.out.println("=".repeat(60));
        System.out.println("  GQS1001 - MATRIZ DE RISCO (Probabilidade x Impacto)");
        System.out.println("=".repeat(60));

        System.out.println("\nEscala de Probabilidade e Impacto (1 a 5):");
        System.out.println("  1 = Muito Baixo | 2 = Baixo | 3 = Médio | 4 = Alto | 5 = Muito Alto");

        List<Funcionalidade> funcionalidades = List.of(
            new Funcionalidade("Checkout/Pagamento", 3, 5),
            new Funcionalidade("Carrinho de Compras", 4, 4),
            new Funcionalidade("Catálogo de Produtos", 3, 2),
            new Funcionalidade("Cadastro de Usuário", 2, 3),
            new Funcionalidade("Busca/Filtros", 4, 2)
        );

        System.out.printf("%n%-25s %-8s %-8s %-8s %-15s%n", "Funcionalidade", "Prob.", "Imp.", "Risco", "Classificação");
        System.out.println("-".repeat(65));
        for (Funcionalidade f : funcionalidades) {
            int risco = calcularRisco(f.probabilidade, f.impacto);
            String classificacao = classificarRisco(risco);
            System.out.printf("%-25s %-8d %-8d %-8d %-15s%n", f.nome, f.probabilidade, f.impacto, risco, classificacao);
        }

        System.out.println("\n--- Priorização de Testes ---");
        System.out.println("  Crítico: Teste exaustivo, automação prioritária");
        System.out.println("  Alto: Teste extensivo");
        System.out.println("  Médio: Teste aprofundado");
        System.out.println("  Baixo: Teste padrão");
    }
}
